public_desktop/desktop-app.js

```diff
@@ -92,6 +94,42 @@ function startDesktopApp() {
 
 	debug = gGebug( 'desktop:browser' );
 
+	try {
+		var buildEditorContextMenu = electron.remote.require( '../desktop/lib/menu/editor-context-menu' );
+		var buildGeneralContextMenu = electron.remote.require( '../desktop/lib/menu/general-context-menu' );
+	} catch (e) {
+		debug( "Error loading context menus", e.message);
+		loadContextMenu = false;
+	}
+	function contextMenu( ev ) {
+		var menu = {};
+		if ( ev.target.closest( 'textarea, input, [contenteditable="true"]' ) ) {
+			menu = buildEditorContextMenu(selection);
+		} else {
+			var selectedText = window.getSelection().toString();
+			menu = buildGeneralContextMenu(selectedText);
+		}
+
+		// The 'contextmenu' event is emitted after 'selectionchange' has fired but possibly before the 
+		// visible selection has changed. Try to wait to show the menu until after that, otherwise the 
+		// visible selection will update after the menu dismisses and look weird. 
+		setTimeout(function() {
+			menu.popup(electron.remote.getCurrentWindow());
+		}, 30);
+	}
+
+	function resetSelection() {
+		selection = {
```

Comment:
FWIW, the name `selection` threw me off since it's different from the general context menu selection.